v4 Redo List of Log Names (#20)
MoonlightCapital/MoonlightBot-docs: advanced/list-of-log-names.md

---

description: On this page all the available logs are listed

---

> # List Of Log Names

Reviewer: ```suggestion
# List of Log Names
```

> Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to include all things that can be logged, including ones that will be introduced in future updates. All logs require the bot to have permission to view and send messages in the logging channel. It must also be able to send embeds and files there,and some logs require the bot to have the "View Audit Log" permission enabled.

Reviewer: ```suggestion
Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [configuring channels](../management-commands/config.md#channels)) are _Capitalized_. Each one of them is separated by categories in the lists below. To log all events, use an asterisk (`*`). All logs require the bot to have permission to view and send messages, embed links and attach files in the respective channels. Some logs require the bot to have the "View Audit Log" permission enabled at server level.
```

| WARNING           | A user [receives a warning](../moderation-commands/warn.md) |

## Members

This category is for actions relating to members interacting with the server.

| Name      | Emitted when     |
| --------- | ---------------- |
| USER_JOIN | Someone joins |
| USER_LEFT | Someone leaves |

## Messages

This category is for things having to do with messages.

| Name                | Emitted when     |
| ------------------- | ---------------- |
| CHANNEL_CLEAN       | Messages get [cleaned](../moderation-commands/clean.md) from a channel. Includes a file with a backup of the deleted messages |
| MESSAGE_BULK_DELETE | A moderator bulk deletes a set of messages |
| MESSAGE_DELETE      | A message is deleted |
| MESSAGE_EDIT        | A message gets edited |
| MESSAGE_PIN_ADD     | A message gets pinned |
| MESSAGE_PIN_REMOVE  | A message gets unpinned |

## Debug

{% hint style="info" %}
This category is for debugging purposes. Logs in this category are meant to aid in fixing errors that may prevent the bot from functioning correctly.

{% endhint %}

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | REACTION_ROLE_DEBUG | An error occurs in the process of triggering a [reaction role.](/start-up/setting-up-reaction-roles.md) |  

Reviewer: ```suggestion
| REACTION_ROLE_DEBUG | An error occurs in the process of triggering a [reaction role](/start-up/setting-up-reaction-roles.md) |  
```

---
description: On this page all the available logs are listed
---

> # [List of log names](list-of-log-names.md#list-of-log-names)

Reviewer: ```suggestion
# List of Log Names
```
Remove hyperlink here

> Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names(which can be used in place of individual log names, see [configuring channels](../admin-commands/config/configuring-channels.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to include all things that can be logged, including ones that will be introduced in future updates.

Reviewer: 1. No space names(which
2. "configuring channels" page link here is incorrect and points to the old format
3. Too little emphasis on the config link anyways, think of someone landing for the first time on this page and thinking "okay the list, what do I do with this?"

> Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to include all things that can be logged, including ones that will be introduced in future updates. All logs require the bot to have permission to view and send messages in the logging channel. It must also be able to send embeds and files there.

Reviewer: The anchor to the channels part goes here after config.md, remove the markdown. I realized now how disconnected the sentence is with the linking to guide part is popping off in the middle.

{% hint style="warning" %}

> Some logs require the bot to have the "View Audit Logs" permission enabled.

Reviewer: Audit Log is singular on Discord

> Some logs require the bot to have the "View Audit Log" permission enabled.

Reviewer: Move all the permission-related part in this box from the paragraph above.

{% endhint %}

## Moderation

> This category is used for actions by moderators or ones that moderators could use. Most of them are fired when a command is run unless they start with AUDIT_. In that case, actions that are done with or without using the bot are logged.

Reviewer: Explains the audit values are taken from the server's audit log

| Name              | Emitted when     |
| ----------------- | ---------------- |

> | BAN               | Someone gets banned |

Reviewer: "someone" in much of these cases is too generic, try using more accurate terms like "A user". Use passive voice too

| INFRACTION_DELETE | Someone removes an infraction|
| KICK              | Someone gets kicked |
| MUTE              | Someone gets muted |

> | PAUSE_ROLE_ADD    | An expiring role gets added  |

Reviewer: "Expiring role"? No, these are for the `/pause-role` command which temporarily removes a role and adds it back at the expiration

| PAUSE_ROLE_EXTEND | A role expiration date gets extended |
| PAUSE_ROLE_REMOVE | A role gets temporarily removed |
| REASON_UPDATE     | A reason for an infraction gets updated |
| ROLE_ADD          | Someone adds a role |
| ROLE_REMOVE       | Someone removes a role|

> | SELFROLE_ADD      | Someone adds a user-assignable role |

Reviewer: self-assignable role, link to the relevant command page

> | REASON_UPDATE     | A [reason for an infraction](../moderation-commands/infractions.md) gets updated |

Reviewer: Link to subcommand # (it's `/infractions reason`)

> | SELFROLE_REMOVE   | A user removes a self-assignable role  |

Reviewer: ```suggestion
| SELFROLE_REMOVE   | A user removes a self-assignable role from themselves  |
```

| SOFTBAN           | A user gets [soft-banned](../moderation-commands/softban.md) |
| TEMPBAN_EXTEND    | A [temporary ban](../moderation-commands/tempban.md) gets extended |
| TEMPBAN_REMOVE    | A temporary ban gets removed |
| TEMPMUTE_EXTEND   | A [temporary mute](../moderation-commands/tempmute.md) gets extended |
| TEMPMUTE_REMOVE   | A temporary ban gets removed |
| TEMPROLE_ADD      | A [temporary role](../role-management-commands/temprole.md) gets added |
| TEMPROLE_EXTEND   | The length of a temporary role gets extended |
| TEMPROLE_REMOVE   | A temporary role gets removed |
| TIMEOUT           | A user gets [timed out](../moderation-commands/timeout.md) |
| TIMEOUT_REMOVE    | A time out gets removed |
| UNBAN             | A user gets [unbanned](../moderation-commands/unban.md) |
| UNMUTE            | A user gets [unmuted](../moderation-commands/unmute.md) |
| WARNING           | A user [receives a warning](../moderation-commands/warn.md) |

## Members

> This category is for when someone leaves or joins the server.

Reviewer: ```suggestion
This category is for actions related to members interacting with the server.
```

| Name      | Emitted when     |
| --------- | ---------------- |
| USER_JOIN | Someone joins |
| USER_LEFT | Someone leaves |

## Messages

This category is for things having to do with messages.

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | CHANNEL_CLEAN       | Someone cleans all the messages from a channel |

Reviewer: Specify it contains a file with the deleted messages

| MESSAGE_BULK_DELETE | Someone bulk deletes a set of messages |

> | MESSAGE_DELETE      | Anyone deletes a message |

Reviewer: ...A message is deleted, passive voice

| MESSAGE_EDIT        | Anyone edits a message |
| MESSAGE_PIN_ADD     | A message gets pinned |
| MESSAGE_PIN_REMOVE  | A message gets unpinned |

## Debug

{% hint style="info" %}
This category is for debugging purposes. Logs in this category are meant to aid in fixing errors that may prevent the bot from functioning correctly.

{% endhint %}

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | REACTION_ROLE_DEBUG | Someone uses a reaction role |

Reviewer: Incorrect, this one is for when someone triggers a reaction role but something goes wrong. It's meant to help the server admins understand what's wring with the server's reaction role configuration.

> | MESSAGE_EDIT        | A user edits a message |

Reviewer: ```suggestion
| MESSAGE_EDIT        | A message gets edited |
```

Consistency with rest of the table

> | REACTION_ROLE_DEBUG | Permissions are misconfigured or someone tries to get a join only role  |

Reviewer: Too specific now, my previous comment to this was only an example for you writer. There are more mishaps triggering the RR debug log

# List Of Log Names

---

> description: On this page all the available logs are listed

Reviewer: No, this desc goes on top between the `---` things like before, likely some GitBook specific meaning

description: On this page all the available logs are listed
---

> # [List Of Log Names](../advanced/list-of-log-names.md#ListOfLogNames)

Reviewer: No! The header must be the first thing on the page. I did not say to swap these two things, I said remove the unnecessary markdown, links, and add the # to the relevant session after config.md

Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to include all things that can be logged, including ones that will be introduced in future updates. All logs require the bot to have permission to view and send messages in the logging channel. It must also be able to send embeds and files there.
# [List Of Log Names](../advanced/list-of-log-names.md#ListOfLogNames)

{% hint style="warning" %}

> Some logs require the bot to have the "View Audit Log" permission enabled.

Reviewer: Why was this removed? I wanted the `All logs require the bot to have permission to view and send messages in the logging channel. It must also be able to send embeds and files there,and some logs require the bot to have the "View Audit Log" permission enabled.` part to be moved inside here to be consistent with all permissions in one place, rephrased if necessary.

# List of log names

> Logs, categorized by what they do are listed here, and are **case sensitive**. Individual logs are written in _UPPER\_CASE_, while category names (which can be used in place of individual log names, see [_**configuring channels**_](../management-commands/config.md)) are _Capitalized_. Each one of them is separated by categories in the lists below. If you want to log everything, use an asterisk to include all things that can be logged, including ones that will be introduced in future updates.

Reviewer: Link this to the # channels anchor to bring the reader to the correct session upon clicking

{% hint style="warning" %}
Some logs require the bot to have the "View Audit Log" permission enabled.
{% endhint %}

## Moderation

> This category is used for actions by moderators or ones that moderators could use. Most of them are fired when a command is run unless they start with AUDIT_, which uses the values in the server's audit log. In that case, actions that are done with or without using the bot are logged.

Reviewer: Wording here is convoluted:

1. You don't "use" an action
2. Data is read from the audit log, "uses" is too generic"
3. "with or without" sounds repetitive

Suggested rewrite:
> This category is to log actions taken by moderators. Most of them are triggered by using bot commands, but those starting with \`AUDIT_\` will read the data from Discord's audit log providing MoonlightBot has permission to do so.

| Name              | Emitted when     |
| ----------------- | ---------------- |
| BAN               | A user gets banned |

> | INFRACTION_DELETE | A moderator removes an infraction|

Reviewer: Deletes is more accurate than removes, links the pages to the relevant (sub)commands to take the actions.

| KICK              | A user gets kicked |
| MUTE              | A user gets muted |
| PAUSE_ROLE_ADD    | A role gets temporarily removed |
| PAUSE_ROLE_EXTEND | A role's length of time to be removed gets adjusted |
| PAUSE_ROLE_REMOVE | A role gets added back |
| REASON_UPDATE     | A reason for an infraction gets updated |
| ROLE_ADD          | A role gets added |
| ROLE_REMOVE       | A role gets removed|

> | SELFROLE_ADD      | A user adds a self-assignable role (see [Selfrole](../role-management-commands/selfrole.md)) |

Reviewer: I prefer the Wikipedia style of linking directly in the words without adding content to make them clickable.

E.g. `A user adds a [self-assignable role](../role-management-commands/selfrole.md)`

And conclude the sentence with "to themselves".

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | CHANNEL_CLEAN       | A moderator cleans all the messages from a channel (which creates a file with the deleted messages) |

Reviewer: Use passive voice, parenthesis is unsound

`Messages get cleaned from a channel. Includes a file with a backup of the deleted messages`

And the link to the relevant page on "cleaned"

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | REACTION_ROLE_DEBUG | Something goes wrong in the process of adding a reaction role |

Reviewer: ```suggestion
| REACTION_ROLE_DEBUG | An error occurs in the process of triggering a reaction role |
```

This is correctly generalized. Now add the link to the tutorial on the words "reaction role"

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | REACTION_ROLE_DEBUG | Someone tries to get a join only role or something else goes wrong in the process of adding a reaction role |

Reviewer: The join only role part is only an example that should be removed from the page.

| Name                | Emitted when     |
| ------------------- | ---------------- |

> | REACTION_ROLE_DEBUG | A user triggers a reaction role, and something goes wtong |

Reviewer: Too vague and there's a typo. This log is triggered, say, when permissions are misconfigured or someone tries to get a join-only role or similar things
